pkg/ipam/crd.go

```diff
@@ -0,0 +1,599 @@
+			nodeResource.ObjectMeta.OwnerReferences = []metav1.OwnerReference{{
+				APIVersion: "v1",
+				Kind:       "Node",
+				Name:       node.GetName(),
+				UID:        k8sNode.UID,
+			}}
+		}
+
+		if option.Config.IPAM == option.IPAMENI {
+			instanceID, instanceType, availabilityZone, vpcID, err := metadata.GetInstanceMetadata()
+			if err != nil {
+				log.WithError(err).Fatal("Unable to retrieve InstanceID of own EC2 instance")
+			}
+
+			nodeResource.Spec.ENI.VpcID = vpcID
+			nodeResource.Spec.ENI.FirstInterfaceIndex = 1
+			nodeResource.Spec.ENI.DeleteOnTermination = true
+			nodeResource.Spec.ENI.PreAllocate = defaults.ENIPreAllocation
+
+			if c := owner.GetNetConf(); c != nil {
+				if c.ENI.MinAllocate != 0 {
+					nodeResource.Spec.ENI.MinAllocate = c.ENI.MinAllocate
+				}
+
+				if c.ENI.PreAllocate != 0 {
+					nodeResource.Spec.ENI.PreAllocate = c.ENI.PreAllocate
+				}
+
+				if c.ENI.FirstInterfaceIndex != 0 {
+					nodeResource.Spec.ENI.FirstInterfaceIndex = c.ENI.FirstInterfaceIndex
+				}
+
+				if len(c.ENI.SecurityGroups) > 0 {
+					nodeResource.Spec.ENI.SecurityGroups = c.ENI.SecurityGroups
+				}
+
+				if len(c.ENI.SubnetTags) > 0 {
+					nodeResource.Spec.ENI.SubnetTags = c.ENI.SubnetTags
+				}
+
+				if c.ENI.VpcID != "" {
+					nodeResource.Spec.ENI.VpcID = c.ENI.VpcID
+				}
+
+				nodeResource.Spec.ENI.DeleteOnTermination = c.ENI.DeleteOnTermination
+			}
+
+			nodeResource.Spec.ENI.InstanceID = instanceID
+			nodeResource.Spec.ENI.InstanceType = instanceType
+			nodeResource.Spec.ENI.AvailabilityZone = availabilityZone
+		}
+
+		_, err := ciliumClient.CiliumV2().CiliumNodes(k8s.DefaultNamespace).Create(nodeResource)
+		if err != nil {
+			log.WithError(err).Error("Unable to create CiliumNode resource")
+		}
+	}
+
+	t, err := trigger.NewTrigger(trigger.Parameters{
+		Name:        "crd-allocator-node-refresher",
+		MinInterval: customResourceUpdateRate,
+		TriggerFunc: store.refreshNodeTrigger,
+	})
+	if err != nil {
+		log.WithError(err).Fatal("Unable to initialize CiliumNode synchronization trigger")
+	}
+	store.refreshTrigger = t
+
+	ciliumNodeSelector := fields.ParseSelectorOrDie("metadata.name=" + nodeName)
+	ciliumNodeStore := cache.NewStore(cache.DeletionHandlingMetaNamespaceKeyFunc)
+	ciliumNodeInformer := informer.NewInformerWithStore(
+		cache.NewListWatchFromClient(ciliumClient.CiliumV2().RESTClient(),
+			"ciliumnodes", v1.NamespaceAll, ciliumNodeSelector),
+		&ciliumv2.CiliumNode{},
+		0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				var valid, equal bool
+				defer func() { owner.K8sEventReceived("CiliumNode", "create", valid, equal) }()
+				if node, ok := obj.(*ciliumv2.CiliumNode); ok {
+					valid = true
+					log.Debugf("New CiliumNode %+v", node)
+					store.updateLocalNodeResource(node)
+					owner.K8sEventProcessed("CiliumNode", "create", true)
+				} else {
+					log.Warningf("Unknown CiliumNode object type %s received: %+v", reflect.TypeOf(obj), obj)
```

Comment:
It's actually intentionally, `WithFields()` does not expand "%+v" the same way. Is there a particular reason why fields are required for this highly unlikely warning that indicates a bug?

Reply:
I was under the impression that we push for the `WithFields()` style so that log gatherers have a better time determining the same log message (similar to our top 5 thing in the CI), and that if the parameters are consistently named, you can do better correlation. The latter may be less applicable here given that the parameters are fairly unique.

Reply:
I've changed all log messages to use fields except for the ones depending on `%+v` behavior.